docs(integrate): Update login users docs (#7414)
zitadel/zitadel: docs/docs/guides/integrate/login/login-users.mdx

This token should not be confused with am access or id tokens in opaque or JWT form that is issued during OpenID connect flows.

:::info Token exchange
Token exchange between Session API and OIDC / SAML tokens is not possible at this moment.
:::

#### Key features of the Session API

These are some key features of the API:

1. **Session Management**: The Session API allows you to create, retrieve, update, and delete sessions for users within your application. Sessions represent a user's active login session, providing temporary access to your application's resources.

2. **Creating Sessions**: With the Session API, you can initiate a new session for a user after they successfully authenticate through ZITADEL. This typically involves generating a session token or ID, which is then associated with the user's identity and used to track their session activity.

3. **Retrieving Sessions**: You can retrieve information about existing sessions using the Session API, such as session ID, user ID, creation time, and expiration time. This allows you to monitor active user sessions and retrieve session details as needed for auditing or troubleshooting purposes.

> 4. **Updating Sessions**: The Session API enables you to update session attributes or extend session lifetimes based on specific application requirements. For example, you might want to refresh a session token periodically to prevent it from expiring prematurely or update session metadata to reflect changes in user permissions or settings.

Reviewer: The main purpose of the update session is, that you can constantly add the verification checks you did. username check, password check, mfa- check. 
Tried to include, maybe you have a better way to write that.

```suggestion
4. **Updating Sessions**: The Session API enables you to update session attributes or extend session lifetimes based on specific application requirements. Each authentication step, such as username / password check or multi-factor verification, will result in an update session. Also you might want to refresh a session token periodically to prevent it from expiring prematurely or update session metadata to reflect changes in user permissions or settings.
```

ZITADEL's hosted login page serves as a centralized authentication interface provided for applications that integrate ZITADEL.
As a developer, understanding the hosted login page is essential for seamlessly integrating authentication into your application.

### Centralized authentication endpoint

ZITADEL's hosted login page acts as a centralized authentication endpoint where users are redirected to authenticate themselves.
When users attempt to access a protected resource within your application, you can redirect them to the hosted login page to authenticate using their login methods and credentials or through Single-sign-on (SSO).
After successful authentication, the user will be redirected back to the originating application.

### Security and compliance

ZITADEL's hosted login page prioritizes security and compliance with industry standards and regulations.
It employs best practices for securing authentication processes, such as encryption, token-based authentication, and adherence to protocols like OAuth 2.0, [OpenID Connect](/docs/guides/integrate/login/oidc), and [SAML](/docs/guides/integrate/login/).

> We make sure to harden your login ui an minimize the attack surface.

Reviewer: ```suggestion
We make sure to harden your login UI an minimize the attack surface.
```

Overall, ZITADEL's hosted login page simplifies the authentication process for developers by providing a secure, customizable, and developer-friendly authentication interface. By leveraging this centralized authentication endpoint, developers can enhance their application's security, user experience, and compliance with industry standards and regulations.

## Key features of the hosted login

### Flexible usernames

Different login name formats can be used on ZITADEL's hosted login page to select a user.
Login methods can be a user's username, containing the username and an [organization domain](/docs/guides/manage/console/organizations#domain-verification-and-primary-domain), their email addresses, or their phone numbers.
By default all of these login methods are allowed and can be adjusted by [Managers](/docs/concepts/structure/managers) to meet their requirements.

### Support for multiple authentication methods

The hosted login page supports various authentication methods, including traditional username/password authentication, social login options, multi-factor authentication (MFA), and passwordless authentication methods like [passkeys](/docs/concepts/features/passkeys.md).

> Available second factor (2FA) and multi-factor authentication methods (MFA) include OTP via authenticator app, OTP via SMS, OTP via email, and U2F.

Reviewer: ```suggestion
Available second factor (2FA) and multi-factor authentication methods (MFA) include OTP via authenticator app, TOTP via SMS, OTP via email, and U2F.
```

Developers can configure the authentication methods offered on the login page based on their application's security and usability requirements.

### Enterprise single-sign-on

> ![Screenshot of zitadel console showing different identity provider templates](/img/guides/integrate/login/login-external-idp-templates.png)

Reviewer: ```suggestion
![Screenshot of ZITADEL console showing different identity provider templates](/img/guides/integrate/login/login-external-idp-templates.png)
```

Key features include:

### Password reset

Unauthenticated users can request a password reset after providing the loginname during the login flow.

- User selects reset password
- An email will be sent to the verified email address
- User opens a link and has to provide a new password

#### Prompt users to setup multifactor authentication

Users are automatically prompted to provide a second factor, when

- Instance or organization [login policy](/concepts/structure/policies#login-policy) is set
- Requested by the client
- A multi-factor is setup for the user

When a multi-factor is required, but not set-up, then the user is requested to set-up an additional factor.

:::info Disabling multifactor prompt
You can disable the prompt, in case multifactor authentication is not enforced by setting the [**Multifactor Init Lifetime**](/docs/guides/manage/console/instance-settings#login-lifetimes) to 0.
:::

#### Enroll passkeys

Users can select a button to initiate passwordless login or use a fall-back method (ie. login with username/password), if available.

The passwordless login flow follows the FIDO2 / WebAuthN standard.
Briefly explained the following happens:

> - User selects button

Reviewer: What is meant with "user selects button"?

Author: removed for now. we probably want to add some video or media on the page.

ZITADEL's hosted login page prioritizes security and compliance with industry standards and regulations.
It employs best practices for securing authentication processes, such as encryption, token-based authentication, and adherence to protocols like OAuth 2.0, [OpenID Connect](/docs/guides/integrate/login/oidc), and [SAML](/docs/guides/integrate/login/).

> We make sure to harden your login UI an minimize the attack surface.

Reviewer 2: ```suggestion
We make sure to harden your login UI and minimize the attack surface.
```

### Support for multiple authentication methods

The hosted login page supports various authentication methods, including traditional username/password authentication, social login options, multi-factor authentication (MFA), and passwordless authentication methods like [passkeys](/docs/concepts/features/passkeys.md).

> Available second factor (2FA) and multi-factor authentication methods (MFA) include OTP via authenticator app, TOTP via SMS, OTP via email, and U2F.

Reviewer 2: ```suggestion
The second factor (2FA) and multi-factor authentication (MFA) methods available in ZITADEL include OTP via the Authenticator app, TOTP via SMS, OTP via email, and U2F.
```

Developers can configure the authentication methods offered on the login page based on their application's security and usability requirements.

### Enterprise single-sign-on

![Screenshot of ZITADEL console showing different identity provider templates](/img/guides/integrate/login/login-external-idp-templates.png)

With the hosted login page from ZITADEL developers will get the best support for multi-tenancy single-sign-on with third party identity providers.

> ZITADEL acts as [identity broker](/docs/concepts/features/identity-brokering) between your applications and different external identity providers, reducing the implementation effort for developers.

Reviewer 2: ```suggestion
ZITADEL acts as an [identity broker](/docs/concepts/features/identity-brokering) between your applications and different external identity providers, reducing the implementation effort for developers.
```
